chore: create package manager template

HonkingGoose opened the pull request:
<!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
<!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->

## Changes:

<!-- Describe what behavior is changed by this PR. -->

- Add new package manager issue template to `docs/development`

## Context:

<!-- Describe why you're making these changes if it's not already explained in a corresponding issue. -->
<!-- If you're closing an existing issue with this pull request, use the keyword Closes #issue_number -->

It might be nice to have a separate template for adding new package managers.

I've linked to our documentation on adding a package manager.

Marking this as draft, as I don't know if you like this, and it needs more work. Best to get feedback before I get too carried away. 😉 

@rarkins found this relevant issue: #5281 (Which proposes a git branch with a "package-manager-template" or even a "scaffolding command").

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

<!-- Do you have any suggestions about this PR template? Edit it here: https://github.com/renovatebot/renovate/edit/main/.github/pull_request_template.md -->

<!-- Please do not force push to your PR's branch after you have created your PR, as doing so forces us to review the whole PR again. This makes it harder for us to review your work because we don't know what has changed. -->
<!-- PRs will always be squashed by us when we merge your work. Commit as many times as you need in this branch. -->

HonkingGoose commented on docs/development/new-package-manager-template.md:
```diff
@@ -0,0 +1,15 @@
+<!-- Please copy/paste this template and fill it in on request from the Renovate maintainers. -->
+
+**What package manager do you want Renovate to support?**
```
Do we want to know if the manager is in wide use or more niche? I sometimes see feature requests to support a quite obscure package manager. Do we intend to support all package managers ever, or do we (want to) have criteria for inclusion?

The reason I'm asking about the criteria is that we incur a support burden for each feature we add, and we will have maintain both code and documentation for each feature. Maybe it might be good to have a better view of "maintainer time vs amount of people we make happy with the feature"?

Let me know your thoughts on this, as I'm curious how you feel about all this. 😉

rarkins replied:
The burden for managers is not too bad, however it would be useful to know how popular it is.

HonkingGoose commented:
@rarkins I've adapted the template you linked in https://github.com/renovatebot/renovate/issues/5281#issuecomment-839536900. It now has checkboxes in some places, and a better heading structure plus `<hr>` elements to break up the sections.

I've added the question about package manager popularity to this template, and added the link to the `docs/development/adding-a-package-manager.md` file.

I'm ready for a first review. 😉

HonkingGoose commented:
@rarkins We could automate "asking to fill out the questionnaire" with the label action we already use.

Proposed workflow:

1. Contributor opens issue asking for new manager support
2. Triagist/maintainers apply the `new manager` label.
3. The label action posts a link to the questionnaire and ask the contributor to fill it in.

Do you like that idea, and if so, do you want a PR from me?
